.github/workflows/pr-ci.yml

```diff
@@ -1,32 +1,44 @@
 name: CI build and push (PR)
 
 concurrency:
-  group: pr-${{ github.event.number }}
+  group: ci-${{ github.run_id }}
   cancel-in-progress: true
 
 on:
-  pull_request_target:
-    types:
-      - opened
-      - reopened
-      - synchronize
-      - labeled
-      - unlabeled
-    branches:
-      - main
-      - v[0-9]+
-      - v[0-9]+.[0-9]+
-      - cryostat-v[0-9]+.[0-9]+
+  issue_comment:
+    types: [created]
 
 jobs:
   code-analysis:
-    uses: ./.github/workflows/ci-code-analysis.yml
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'cryostatio' && contains(github.event.pull_request.labels.*.name, 'safe-to-test') && github.event.issue.pull_request && startsWith(github.event.comment.body, '/build_test')
+    steps:
+    - name: React to comment
+    uses: actions/github-script@v4
     with:
-      checkout-repo: ${{ github.event.pull_request.head.repo.full_name }}
-      checkout-ref: ${{ github.event.pull_request.head.ref }}
-    secrets: inherit
-    if: github.repository_owner == 'cryostatio' && contains(github.event.pull_request.labels.*.name, 'safe-to-test')
-
+      script: |
+          const {owner, repo} = context.issue
+          github.reactions.createForIssueComment({
+            owner,
+            repo,
+            comment_id: context.payload.comment.id,
+            content: "+1",
+          });  
+    - uses: tspascoal/get-user-teams-membership@v2
+      id: checkMember
+      with:
+        username: ${{ github.actor }}
+        team: 'cryostat/reviewers'
+        GITHUB_TOKEN: ${{ secrets.PAT }}
```

Comment:
I think we can elevate the permissions of this token for action to allow querying organization?

Reply:
Also, its worth exploring if that can extend to the use of pushing images to `ghcr.io`. That way, we can use `GITHUB_TOKEN` and eliminate the need for `target_pull_target`.

Reply:
the token needs read:org scope so I think users need to grant it.... I'm not certain though

Reply:
@tthvo

Reply:
Sorry to clarify, i mean we could, instead of using PAT, use the `GITHUB_TOKEN` with elevated permissions.

Check this out: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

Reply:
Unfortunately, there is no option for org query:

```yaml
permissions:
  actions: read|write|none
  checks: read|write|none
  contents: read|write|none
  deployments: read|write|none
  id-token: read|write|none
  issues: read|write|none
  discussions: read|write|none
  packages: read|write|none
  pages: read|write|none
  pull-requests: read|write|none
  repository-projects: read|write|none
  security-events: read|write|none
  statuses: read|write|none
```

Reply:
I mention this because we will run into issues with accessing repo secrets if the event is `issue_comment`.

Reply:
Well there could be another way:

- `safe-to-test` label: Enable the PR author to run the `/build_test` command.
- `auto-test`: A new label to do behave as what the `safe-to-test` label does now.

This way, we don't have to ensure the commenter is the cryostat reviewer. The author can just do it themselves if granted/trusted -> save us the need for PAT.